source/uri.c

```diff
@@ -375,17 +377,23 @@ static void s_parse_authority(struct uri_parser *parser, struct aws_byte_cursor
         }
 
         const uint8_t *port_delim = memchr(port_search_start, ':', port_search_len);
-
+        /*
+         * RFC-3986 section 3.2.2: A host identified by an IPv6 literal address is represented inside the square
+         * brackets.
+         * Ignore the sqaure brackets.
+         */
+        size_t host_name_offset = is_IPv6_literal ? 1 : 0;
+        size_t host_name_length_correction = is_IPv6_literal ? 2 : 0;
         if (!port_delim) {
             parser->uri->port = 0;
-            parser->uri->host_name = authority_parse_csr;
+            parser->uri->host_name.ptr = authority_parse_csr.ptr + host_name_offset;
+            parser->uri->host_name.len = authority_parse_csr.len - host_name_length_correction;
```

Comment:
ugh, not a big fan of manual cursor manipulation? 
assign authority_parse_csr to host and then advance and dec len by 1 to remove brackets?

Reply:
Thanks, I have updated the code to advance and decrement the length. Initially, I wrote the following code: https://github.com/awslabs/aws-c-common/pull/1112/commits/1127cad9dfc3a730647b27dd238bb1dbe01236c4, but later refactored it to reduce code duplication. We can also refactor that version to use 'advance' and 'decrement length.' If you like that version better, I am conflicted between them.

Reply:
thanks. i think intent of what you are trying to do is much clearer in the latest version.

one further cleanup we could potentially do is to use aws_byte_cursor_next_split to separate host from port in authority. we can probably even get rid of memchr using that approach. but its a bigger refactoring, so we can do it in future